File: UserInputService/Producer/producer.py
from confluent_kafka import KafkaError
from confluent_kafka import Producer


class MyProducer:
    producer_config = {
        'bootstrap.servers': 'localhost:9092',
    }
    
    def __init__(self, topic) -> None:
        self.producer = Producer(
            self.producer_config
        )

        self.topic = topic

    def produce(self, value):
        #Produces message to kafka topic
        self.producer.produce(topic=self.topic, value=value)
        self.producer.flush()

File: GrammarChecker/Producer/producer.ts
import { kafka } from ".."



export class MyProducer {
    private readonly _topic = 'grammar-checked';

    //Produce a message to kafka topic.
    async produce(message: string) {

        const producer = kafka.producer()
        await producer.connect()
        await producer.send({
            topic: this._topic,
            messages: [
                {value: message}
            ]
        });

        await producer.disconnect();




    }



}